### ISSUE TITLE
false positive on main's argc / argv

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#43061]
|  |  |
| --- | --- |
| Bugzilla Link | [43716](https://llvm.org/bz43716) |
| Version | unspecified |
| OS | Linux |
| Attachments | [clang report](https://user-images.githubusercontent.com/60944935/143759940-b2e3f55b-1840-459a-bab6-1f56b85acd8f.gz) |
| Reporter | LLVM Bugzilla Contributor |
| CC | @zygoloid |

## Extended Description 

I can not simplify the code.

probably connection between `argc` and `argv` is lost after
```cpp
char **path = argv + 3;
```

**Comment by llvmbot** [comment]
@llvm/issue-subscribers-clang-static-analyzer

Author: None (llvmbot)

<details>
|  |  |
| --- | --- |
| Bugzilla Link | [43716](https://llvm.org/bz43716) |
| Version | unspecified |
| OS | Linux |
| Attachments | [clang report](https://user-images.githubusercontent.com/60944935/143759940-b2e3f55b-1840-459a-bab6-1f56b85acd8f.gz) |
| Reporter | LLVM Bugzilla Contributor |
| CC | @zygoloid |

## Extended Description 


I can not simplify the code.

probably connection between argc and argv is lost after

char **path = argv + 3;
</details>
